src/core/modal.slice.ts

```diff
@@ -10,6 +10,17 @@ interface PayloadProps {
 interface StateProps {
   modalIds: string[];
 }
+const focusContainerArray: Element[] = [];
+const handleModalFocus = (elementToStore?: Element) => {
+  if (elementToStore) {
+    return focusContainerArray.push(elementToStore);
+  }
+  const element = focusContainerArray.pop() as HTMLElement;
+  if (element) {
+    element.focus();
+  }
+  return element;
+};
```

Comment:
I like the idea of it but I think you could split the function in two functions (names are just suggestions):
SetElementFocus and FocusLatestElement
Just to make clear what is happening.

Reply:
Another question:
What is the idea behind that it is an array?
Why can there be multiple elements in the history?
Would it be more safe and less prone to errors to just have one element?

Reply:
Some modals lead into other modals, which means that we potentially can have x modals open on top of each other at a time. We need to store an array of elements in the order that the respective modals were opened, in order to return the correct element into focus when a modal is closed.
 
I get the idea of splitting the functionality into two seperate functions. One for storing and one for returning. I'll look into that.

Reply:
> Some modals lead into other modals, which means that we potentially can have x modals open on top of each other at a time. We need to store an array of elements in the order that the respective modals were opened, in order to return the correct element into focus when a modal is closed.

Ok. Thank you for the explanation!

> I get the idea of splitting the functionality into two seperate functions. One for storing and one for returning. I'll look into that.

👍